attestation/k8smanifest/k8smanifest.go

```diff
@@ -0,0 +1,485 @@
+package k8smanifest
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"math"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/in-toto/go-witness/attestation"
+	"github.com/in-toto/go-witness/cryptoutil"
+	"github.com/in-toto/go-witness/log"
+	"github.com/in-toto/go-witness/registry"
+	"github.com/invopop/jsonschema"
+)
+
+// Name is the identifier for this attestor.
+const Name = "k8smanifest"
+
+// Type is the URI identifying the predicate type.
+const Type = "https://witness.dev/attestations/k8smanifest/v0.1"
+
+// RunType is the run stage at which this attestor is executed.
+const RunType = attestation.PostProductRunType
+
+// Default ephemeral fields to remove.
+var defaultEphemeralFields = []string{
+	"metadata.resourceVersion",
+	"metadata.uid",
+	"metadata.creationTimestamp",
+	"metadata.managedFields",
+	"metadata.generation",
+	"status",
+}
+
+// Default ephemeral annotations to remove.
+var defaultEphemeralAnnotations = []string{
+	"kubectl.kubernetes.io/last-applied-configuration",
+	"deployment.kubernetes.io/revision",
+	"witness.dev/content-hash",
+	"cosign.sigstore.dev/message",
+	"cosign.sigstore.dev/signature",
+	"cosign.sigstore.dev/bundle",
+}
+
+// RecordedObject stores ephemeral-cleaned doc details.
+type RecordedObject struct {
+	FilePath       string                 `json:"filePath"`
+	Kind           string                 `json:"kind"`
+	Name           string                 `json:"name"`
+	Data           map[string]interface{} `json:"data"`
+	CanonicalJSON  string                 `json:"canonicalJSON"`
+	SubjectKey     string                 `json:"subjectKey"`
+	ComputedDigest cryptoutil.DigestSet   `json:"computedDigest"`
+}
+
+// Attestor implements the Witness Attestor interface for Kubernetes manifests.
+type Attestor struct {
+	ServerSideDryRun  bool     `json:"server_side_dry_run,omitempty"`
+	KubeconfigPath    string   `json:"kubeconfig,omitempty"`
+	IgnoreFields      []string `json:"IgnoreFields,omitempty" jsonschema:"title=IgnoreFields"`
```

Comment:
Should we be standardizing to a specific format for the json keys (e.g., `snake_case, camelCase, PascalCase`?). Just gone and checked and it seems we don't have standardization across our attestors. Would be nice to have.

Reply:
that is going to be a mess.  What do we want to do? This struct was generated, so I have no preference.  cc @mikhailswift